lib/src/decorator.dart

```diff
@@ -0,0 +1,150 @@
+
+String _getBrowserVersionClasses() {
+  var majorVersion = browser.version.major;
+  var classes = [
+    _browserClassPrefix + _browserClassName + majorVersion.toString()
+  ];
+
+  for (var i = minVersionRange; i < minVersionRange + maxVersionRange; i++) {
+    classes.add(
+        '$_browserClassPrefix$_versionRangeClassPrefix$_browserClassName$i');
+  }
+
+  return _listToClassNameString(classes);
+}
+
+String _getFeatureSupportClasses(List<Feature> features) {
+  var classes = [];
+
+  for (var i = 0; i < features.length; i++) {
+    var feature = features[i];
+
+    classes.add(feature.isSupported
+        ? feature.name
+        : '${_featureSupportNegationClassPrefix}${feature.name}');
+  }
+
+  return _listToClassNameString(classes);
+}
+
+/// Utility fn that takes a list of [classes] and returns a space-separated string for
+/// use within the `class` attribute on the `<html>` element that gets injected by
+/// [decorateRootNodeWithPlatformClasses].
+String _listToClassNameString(List<String> classes) {
+  var classesStr = '';
+
+  for (var i = 0; i < classes.length; i++) {
+    if (classesStr.isEmpty) {
+      classesStr = classes[i];
+    } else {
+      classesStr += ' ${classes[i]}';
+    }
+  }
+
+  return classesStr;
+}
+
+/// Convert white-space within the given [name] to dashes, and convert it to lowercase
+/// for standardized CSS class formatting.
+@visibleForTesting
+String nameToClassName(String name) {
+  return name.replaceAll(' ', '-').toLowerCase();
+}
+
+/// Whether the given [rootNode] has already had it's CSS classes set via [decorateRootNodeWithPlatformClasses].
+bool nodeHasBeenDecorated(Element rootNode) =>
+    rootNode.classes.contains(_decorationCompleteClassName);
+
+/// Generates CSS classes based on the current [browser], [operatingSystem] and optionally,
+/// [features] that your app may need conditional styling for in addition to the
+/// [defaultFeatureCssClassDecorators] that will have CSS classes present by default.
+String getPlatformClasses({List<Feature> features, List<String> decorators}) {
+  decorators ??= [];
+
+  if (features != null && features.isNotEmpty) {
+    features.addAll(defaultFeatureCssClassDecorators);
+  } else {
+    features = defaultFeatureCssClassDecorators;
+  }
+
+  decorators.add(_browserClassPrefix + _browserClassName);
+  decorators.add(_getBrowserVersionClasses());
+  decorators.add(_osClassPrefix + _osClassName);
+  decorators.add(_getFeatureSupportClasses(features));
+  decorators.add(_decorationCompleteClassName);
+
+  return _listToClassNameString(decorators);
+}
+
+/// Appends CSS classes generated by [getPlatformClasses] to the specified [rootNode].
+///
+/// By default, [rootNode] is [document.documentElement].
+void decorateRootNodeWithPlatformClasses(
+    {List<Feature> features, Element rootNode, Function callback}) {
+  if (document == null) return;
+
+  rootNode ??= document.documentElement;
+
+  if (rootNode != null && !nodeHasBeenDecorated(rootNode)) {
```

Comment:
I think there could be cases in wdesk where multiple packages/experiences are running and using `platform_detect`, and relying on different features. Currently only the first call to `decorateRootNodeWithPlatformClasses` will result in the classes being added. I know we don't want to cause duplication of the feature class names - could we change `getPlatformClasses` to add all the classes into a `Set` instead of a `List`?